x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -862,7 +862,7 @@ public void testCombineOrderByThroughProjectAndEval() {
 
         var keep = as(plan, Project.class);
         var topN = as(keep.child(), TopN.class);
-        assertThat(orderNames(topN), contains("salary", "emp_no"));
```

Comment:
I am not convinced this change is correct either.
We must not forget that there is always a `limit` pushed down, be it the one the user provided or one of the current defaults (500, 10000). A query like

```
from test
            | sort emp_no
            | rename emp_no as en
            | keep salary, en
            | eval e = en * 2
            | sort salary
```

must sort by `emp_no` and keep 500 documents. Afterwards comes the sorting by `salary` which will be different if there is no sorting by `emp_no` first.

Did I understand right that `sort emp_no` disappears completely from the plan?

Reply:
```
from test
| sort emp_no
| rename emp_no as en
| keep salary, en
| eval e = en * 2
| sort salary
```

becomes

```
from test
| sort emp_no
| rename emp_no as en
| keep salary, en
| eval e = en * 2
| sort salary
| limit 500
```
which becomes

```
from test
| rename emp_no as en
| keep salary, en
| eval e = en * 2
| sort salary
| limit 500
```

as the first sort is removed resulting in topN(salary, 500)

Reply:
That wasn't the detail that I was missing, but how much the `limit` is being pushed down. And it's not pushed passed `sort salary`. All good.